plugin.yaml

```diff
@@ -128,94 +128,158 @@ data_types:
         required: true
       sourcePortRange:
+        description: |
+          Single port number from 1 to 65535,
+          port range (i.e. 1-65635),
+          or * (for all ports)
         type: string
         required: false
       destinationPortRange:
+        description: |
+          Single port number from 1 to 65535,
+          port range (i.e. 1-65635),
+          or * (for all ports)
         type: string
         required: false
       sourceAddressPrefix:
+        description: |
+          Single IP address (i.e. 10.10.10.10),
+          IP subnet (i.e. 192.168.1.0/24),
+          default tag,
+          or * (for all addresses)
         type: string
         required: true
       destinationAddressPrefix:
+        description: |
+          Single IP address (i.e. 10.10.10.10),
+          IP subnet (i.e. 192.168.1.0/24),
+          default tag,
+          or * (for all addresses)
         type: string
         required: true
       access:
+        description: allow or deny
         type: string
         required: true
       priority:
+        description: |
+          Number between 100 and 4096
         type: integer
         required: true
       direction:
+        description: inbound or outbound
         type: string
         required: true
-        
+
   cloudify.datatypes.azure.network.SubnetConfig:
     description: >
       See https://msdn.microsoft.com/en-us/library/mt163621.aspx
     properties:
       addressPrefix:
+        description:
+          The address prefix to use
         required: false
       networkSecurityGroup:
+        description:
+          The name of a security group to attach, if desired.
         required: false
       routeTable:
+        description:
+          The name of a route table to attach, if desired.
         required: false
-        
+
   cloudify.datatypes.azure.network.RouteTableConfig:
     description: >
       See https://msdn.microsoft.com/en-us/library/mt502548.aspx
     properties:
       routes:
         required: false
-        
+        description: |
+          List of route definitions.
+          (see :cfy:node`cloudify.azure.nodes.network.Route`).
+
   cloudify.datatypes.azure.network.RouteConfig:
     description: >
       See https://msdn.microsoft.com/en-us/library/mt459110.aspx
     properties:
       addressPrefix:
+        description:
+          The destination CIDR to which the route applies
         type: string
         required: true
       nextHopType:
+        description:
+          The type of Azure hop the packet should be sent to
         type: string
         required: true
       nextHopIpAddress:
+        description:
+          The IP address packets should be forwarded to
         type: string
         required: false
-        
+
 
   cloudify.datatypes.azure.network.NetworkInterfaceCardConfig:
     description: >
       See https://msdn.microsoft.com/en-us/library/mt163668.aspx
     properties:
       networkSecurityGroups:
```

Comment:
Based on the linked API doc this should actually be `networkSecurityGroup`? Unless it's being handled by the plugin somewhere (I didn't look in depth)?